tls/s2n_config.c

```diff
@@ -284,7 +284,10 @@ struct s2n_config *s2n_config_new(void)
     GUARD_PTR(s2n_alloc(&allocator, sizeof(struct s2n_config)));
 
     new_config = (struct s2n_config *)(void *)allocator.data;
-    GUARD_PTR(s2n_config_init(new_config));
+    if (s2n_config_init(new_config)) {
```

Comment:
Can you make this an explicit comparison to `!= S2N_SUCCESS`, instead? Makes it a bit easier to read.